### DISCUSSION TITLE
syntax for multiple caption in a single line

### DISCUSSION THREAD

**Discussion by Pierre9344** [quarto-dev/quarto-cli#2520] [Q&A] [answered]
Hello

Is it possible to indicate figure caption of multiple figure in a single line with a syntax similar to the one used for the tabset?

I'm creating a document that iterate a list to create tab and figure. I'm wondering if it is possible to indicate the captions in the same way I indicate the `::: panel-tabset` ? I precise that the plot(cluster_model) return 8 plot each time I call it.

Thank you in advance for your answer.

````
```{r}
#| results: 'asis'
for (i in c("ALL", "A", "B", "C")) {
  # create tab for each groups of individuals and prepare the sub-tabs for the different clustering on this group
  cat("### ", i, "\n::: panel-tabset\n")
  for (j in rownames(ls)) {
    if (!is.na(ls[j, i])) {
      # check if the file exist
      cat("#### ", j, "\n") # create the clustering sub-tabs
      load(ifelse(i == "ALL",
        paste0("./data/base/proposalClassification/4visits/", permut, " permut/", ls[j, i]),
        paste0("./data/base/proposalClassification/4visits/", permut, " permut/", i, "/", ls[j, i])
      ))
      # get the plots, the layout-ncol div is used to get 2 plots on the same line
      cat("\n::: {layout-ncol=2}\n")
      plot(cluster_model)
      cat("\n:::\n")
    }
  }
  cat("\n:::\n")
}
```
````

**Comment by cderv** [comment, marked as answer]
I believe you should iterate on your content by leveraging child document in which you can fully right usual Qmd syntax. This will be safer for the outputs, and easier for using **knitr** and Quarto feature such as caption. 

See my answer in this thread (https://github.com/quarto-dev/quarto-cli/issues/2370#issuecomment-1241946395) and this one (https://github.com/quarto-dev/quarto-cli/discussions/2403#discussioncomment-3628538) regarding how to correctly iterate to program the creation of mix of Markdown + Code content. 

Regarding figure caption for multiple plot, **knitr** should handle that using vector for figure options. This should work within Quarto too

````markdown
---
format: html
---

```{r lab}
#| layout-ncol: 2
#| fig-cap: 
#|   - mtcars
#|   - cars
plot(mtcars)
plot(cars)
```
```` 
![image](https://user-images.githubusercontent.com/6791940/191711646-2791a247-bf3a-4ffa-8b93-4e03753fda50.png)

You should try for only one clustering group to see how it looks likes and if it works. I am not sure how the `plot()` method you are using works as you did not share reproducible code (missing data and packages used).
